assets/src/edit-story/app/media/local/provideContextValue.js

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useEffect, useCallback, useRef } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useAPI, useConfig } from '../..';
+import useUploadVideoFrame from '../utils/useUploadVideoFrame';
+import useUploadMedia from '../useUploadMedia';
+import { getResourceFromAttachment } from '../utils';
+
+/* eslint-disable react-hooks/rules-of-hooks */
```

Comment:
provideContextValue is clearer than the alternatives, eg:
- useContextValue() - may be confused with useContext(), which is a consumer.
- useContextValueProvider() - doesn't return a provider fn, the value itself is returned.
- useProvideContextValue() - doesn't form a valid sentence.

Reply:
This has been renamed to be useContextValueProvider.